media/transcoder/create_job_with_periodic_images_spritesheet.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Google Cloud Transcoder sample for creating a job that generates two spritesheets from the input video. Each spritesheet contains images that are captured periodically.
+
+Example usage:
+    python create_job_with_periodic_images_spritesheet.py --project-id <project-id> --location <location> --input-uri <uri> --output-uri <uri>
+"""
+
+import argparse
+
+from google.cloud.video import transcoder_v1beta1
+from google.cloud.video.transcoder_v1beta1.services.transcoder_service import (
+    TranscoderServiceClient,
+)
+from google.protobuf import duration_pb2 as duration
+
+# [START transcoder_create_job_with_periodic_images_spritesheet]
+
+
+def create_job_with_periodic_images_spritesheet(
+    project_id, location, input_uri, output_uri
+):
+    """Creates a job based on an ad-hoc job configuration that generates two spritesheets.
+
+    Args:
+        project_id: The GCP project ID.
+        location: The location to start the job in.
+        input_uri: Uri of the video in the Cloud Storage bucket.
+        output_uri: Uri of the video output folder in the Cloud Storage bucket."""
+
+    client = TranscoderServiceClient()
+
+    parent = f"projects/{project_id}/locations/{location}"
+    job = transcoder_v1beta1.types.Job()
+    job.input_uri = input_uri
+    job.output_uri = output_uri
+    job.config = transcoder_v1beta1.types.JobConfig(
+        elementary_streams=[
+            transcoder_v1beta1.types.ElementaryStream(
+                key="video-stream0",
+                video_stream=transcoder_v1beta1.types.VideoStream(
+                    codec="h264",
+                    height_pixels=360,
+                    width_pixels=640,
+                    bitrate_bps=550000,
+                    frame_rate=60,
+                ),
```

Comment:
Presumably, these are all specific to the URI for the video used for testing? Can you add some comments to indicate that the user should be updating these values to match the video submitted, and if there are any constraints on what they might be? 

There's also quite a bit of nesting here, which [we generally try to avoid having too much of](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#arrange-act-assert). It would be good to pull maybe the top level components out so it context can be added?

Reply:
These attributes are for the output video. I added some comments to clarify, but these particular samples focus on the spritesheet functionality.

This nesting is somewhat unavoidable. The job config is lengthy. I added further comments to clarify the functions of each section, but not sure pulling out the top-level components will help and may just add unnecessary length.